milli/src/vector/mod.rs

```diff
@@ -26,6 +30,171 @@ pub type Embedding = Vec<f32>;
 
 pub const REQUEST_PARALLELISM: usize = 40;
 
+pub struct ArroyWrapper {
+    quantized: bool,
+    index: u16,
+    database: arroy::Database<Unspecified>,
+}
+
+impl ArroyWrapper {
+    pub fn new(database: arroy::Database<Unspecified>, index: u16, quantized: bool) -> Self {
+        Self { database, index, quantized }
+    }
+
+    pub fn index(&self) -> u16 {
+        self.index
+    }
+
+    pub fn dimensions(&self, rtxn: &RoTxn) -> Result<usize, arroy::Error> {
+        if self.quantized {
+            Ok(arroy::Reader::open(rtxn, self.index, self.quantized_db())?.dimensions())
+        } else {
+            Ok(arroy::Reader::open(rtxn, self.index, self.angular_db())?.dimensions())
+        }
+    }
+
+    pub fn quantize(
+        &mut self,
+        wtxn: &mut RwTxn,
+        index: u16,
+        dimension: usize,
+    ) -> Result<(), arroy::Error> {
+        if !self.quantized {
+            let writer = arroy::Writer::new(self.angular_db(), index, dimension);
+            writer.prepare_changing_distance::<BinaryQuantizedAngular>(wtxn)?;
+            self.quantized = true;
+        }
+        Ok(())
+    }
+
+    pub fn need_build(&self, rtxn: &RoTxn, dimension: usize) -> Result<bool, arroy::Error> {
```

Comment:
Is a build needed when `quantize` was called, or is the following sequence of uses valid?

1. Self::quantize
2. Self::needs_build -> returns false
3. Self::nns_by_item()

Reply:
> Is a build needed when quantize was called

Yes! And we now have a test on this in arroy 😎